src/sirfilesystem.c

```diff
@@ -77,13 +77,26 @@ bool _sir_pathgetstat(const char* restrict path, struct stat* restrict st, sir_r
 #  error "unknown open_flags for your platform; please contact the developers."
 # endif
 
+# if !(defined(_AIX) || !defined(__PASE__))
         int fd = open(base_path, open_flags);
         if (-1 == fd) {
             _sir_safefree(&base_path);
             return _sir_handleerr(errno);
         }
-
         stat_ret = fstatat(fd, path, st, AT_SYMLINK_NOFOLLOW);
+# else
+        // HACKHACK: fstatat does not work properly for any fd other than AT_FDCWD.
+        SIR_UNUSED(open_flags);
+        int fd = AT_FDCWD;
+        if (rel_to == SIR_PATH_REL_TO_CWD) {
+            /* cppcheck-suppress invalidFunctionArg */
```

Comment:
Where is there an invalid arg error from cppcheck?

Reply:
I'll remove it to allow it to fail.